compiler/noirc_evaluator/src/ssa/ssa_gen/program.rs

```diff
@@ -98,3 +108,45 @@ impl Display for Ssa {
         Ok(())
     }
 }
+
+#[cfg(test)]
+mod test {
+    use crate::ssa::ir::map::Id;
+
+    use crate::ssa::ssa_gen::Ssa;
+    use crate::ssa::{
+        function_builder::FunctionBuilder,
+        ir::{instruction::BinaryOp, types::Type},
+    };
+
+    #[test]
+    fn serialization_roundtrip() {
+        // fn main f0 {
+        //   b0(v0: Field):
+        //     v1 = add v0, Field 1
+        //     v2 = mul v1, Field 3
+        //     return v2
+        // }
+        let main_id = Id::test_new(0);
+
+        // Compiling main
+        let mut builder = FunctionBuilder::new("main".into(), main_id);
+        let v0 = builder.add_parameter(Type::field());
+
+        let one = builder.field_constant(1u128);
+        let three = builder.field_constant(3u128);
+
+        let v1 = builder.insert_binary(v0, BinaryOp::Add, one);
+        let v2 = builder.insert_binary(v1, BinaryOp::Mul, three);
+        builder.terminate_with_return(vec![v2]);
+
+        let ssa = builder.finish();
+        let expected_string = format!("{}", ssa);
```

Comment:
Can you write the actual String expected here instead of programmatically generating it? It could be good to replace the comment on line 124.

Reply:
I switched to use the actual String expected here. It was easiest to include the string with the newline characters and spaces as otherwise formatting starts to mess up the expected string.